### PR TITLE
feat(wallet): dedicated aggregated transactions page

### PR DESCRIPTION
**Pull request by josheleonard**
Resolves https://github.com/brave/brave-browser/issues/27106

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

Displays an informational message when there are no transactions to display
Screenshot 2022-12-19 at 3 55 18 PM

Search box correctly filters the transactions list

https://user-images.githubusercontent.com/30185185/208547862-d1324519-834b-474b-8d2b-6a15e7ed0612.mov

Displays transactions once they are available 

https://user-images.githubusercontent.com/30185185/208548112-9a4571c8-a176-47db-8b11-6726def6cfed.mov

https://user-images.githubusercontent.com/30185185/208548123-da3ba48f-282f-4406-8296-927d5a29fce8.mov

https://user-images.githubusercontent.com/30185185/208548148-e4aaf10e-4458-4e40-bab6-8fc8a9fd4f2c.mov

Automatically displays updates to transaction statuses

https://user-images.githubusercontent.com/30185185/208548243-d1124cb3-5b29-4c30-9ebc-7f497baa176c.mov

### CODE REVIEW

File: components/brave_wallet_ui/components/desktop/portfolio-transaction-item/index.tsx [hunk 1]
```diff
@@ -272,76 +301,75 @@ export const PortfolioTransactionItem = React.forwardRef<HTMLDivElement, Props>(
...
+          {
+            transaction.txType === BraveWallet.TransactionType.ERC721TransferFrom ||
+              transaction.txType === BraveWallet.TransactionType.ERC721SafeTransferFrom
+              ? ' ' + transaction.erc721TokenId
+              : ''
+          }
+        </AddressOrAsset>
+      </>
+    )
+  }, [transaction, transaction, displayAccountName, onAssetClick])
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/portfolio-transaction-item/index.tsx:360]: `transaction` is added twice here.

File: components/brave_wallet_ui/common/hooks/transaction-parser.ts [hunk 2]
```diff
@@ -150,67 +164,49 @@ export function useTransactionFeesParser (selectedNetwork?: BraveWallet.NetworkI
...
-        normalizedTransferredValue,
-        spotPrices,
-        tx,
-        sellToken,
-        token,
-        txNetwork: selectedNetwork,
-        transferredValueWei: weiTransferredValue
-      })
-    }
   }, [
```
[the remark this answers is not in the document]
Comment (josheleonard) [reply components/brave_wallet_ui/common/hooks/transaction-parser.ts:195]: added missing deps

File: components/brave_wallet_ui/components/desktop/portfolio-transaction-item/index.tsx (hunk 1 above)
Comment (josheleonard) [reply components/brave_wallet_ui/components/desktop/portfolio-transaction-item/index.tsx]: removed duplicate hook dependency arg

**Review by Douglashdaniel — APPROVED** [review]
++

**Pull request merged**